node/cmd/ccq/permissions.go

```diff
@@ -114,26 +115,46 @@ func NewPermissions(fileName string, env common.Environment) (*Permissions, erro
 }
 
 // StartWatcher creates an fswatcher to watch for updates to the permissions file and reload it when it changes.
-func (perms *Permissions) StartWatcher(ctx context.Context, logger *zap.Logger, errC chan error) {
+func (perms *Permissions) StartWatcher(ctx context.Context, logger *zap.Logger, errC chan error) error {
 	logger = logger.With(zap.String("component", "perms"))
-	perms.watcher = fswatch.NewWatcher(perms.fileName)
-	fsChan := perms.watcher.Start()
+	permWatcher, createErr := fsnotify.NewWatcher()
+	if createErr != nil {
+		return createErr
+	}
+
+	// fsnotify requires watching the parent directory rather than the file.
+	watchDir := path.Dir(perms.fileName)
+	addErr := permWatcher.Add(watchDir)
+	if addErr != nil {
+		return addErr
+	}
+	perms.watcher = permWatcher
+	logger.Warn("Starting permissions watcher", zap.String("dir", watchDir))
 
 	common.RunWithScissors(ctx, errC, "perm_file_watcher", func(ctx context.Context) error {
 		for {
 			select {
 			case <-ctx.Done():
+				logger.Info("got context done")
 				return nil
-			case notif := <-fsChan:
-				if notif.Path != perms.fileName {
-					return fmt.Errorf("permissions watcher received an update for an unexpected file: %s", notif.Path)
+			case event := <-perms.watcher.Events:
+				logger.Debug("got permissions watcher event", zap.String("event", event.String()))
+
+				// Look for modifications to the permissions file. Which event is triggered
+				// depends on how the file was modified: something like nano will issue
+				// a Write event, where Vim actually deletes the file and recreates it on save.
+				//
+				// NOTE: A `touch` command issues only a `Chmod` event,
+				// so it will trigger this branch.
```

Comment:
Sorry, did you mean "will not" rather than "will"? Just wanted to clarify.

Reply:
Yes I did!
```suggestion
				// so it will not trigger this branch.
```